crypto/mem_sec.c

```diff
@@ -137,11 +137,17 @@ void *CRYPTO_secure_malloc(size_t num, const char *file, int line)
 
 void *CRYPTO_secure_zalloc(size_t num, const char *file, int line)
 {
+#ifdef IMPLEMENTED
+    if (!secure_mem_initialized)
+        return CRYPTO_zalloc(num, file, line);
+    return CRYPTO_secure_malloc(num, file, line);
```

Comment:
This seems reasonable.  What about: `/* CRYPTO_secure_malloc() now zeroes allocations when secure memory is available */`.  I'd rather not mislead anyone about it always clearing memory, it doesn't.  I contemplated using a value other than zero for the `memset(3)` to avoid accidentally initialising things for callers.

Reply:
I don't get this: if this is applied `CRYPTO_secure_zalloc()` will not clear the whole allocated memory, right?

I don't think that's what the developer expects!

Reply:
> The buffer will be cleared on free and is cleared after the initial mmap(2). Thus, it will always be zero unless it is allocated (once the free list header is dealt with). So, there should be no need to clear the entire block of memory before returning as @mspncp pointed out.

I see now! Uhm, maybe it's even better if this was documented in the comments to avoid confusion!